tests/integration/replica-redirect.tcl

```diff
@@ -32,5 +37,42 @@ start_server {tags {needs:repl external:skip}} {
             r readonly
             r get foo
         } {}
+
+        test {client paused during failover-in-progress} {
+            pause_process $replica_pid
+            # replica will never acknowledge this write
+            r -1 set foo bar
+            r -1 failover to $replica_host $replica_port TIMEOUT 100 FORCE
```

Comment:
@soloestoy This tests seems to be consistently failing under valgrind. We might need some better checks to make sure the node is online before continuing here. https://github.com/valkey-io/valkey/actions/runs/10396494542/job/28794814216

Reply:
@madolson @soloestoy Proposed fix is at #917